tests/functional/lib/helpers.js

```diff
@@ -53,103 +53,113 @@ define([
       return this.parent
         .then(function () {
           if (options.contentServer) {
-            return clearContentServerState(this.parent, options);
+            return this.parent
+              .then(clearContentServerState(options));
           }
         })
         .then(function () {
           if (options['123done']) {
-            return clear123DoneState(this.parent);
+            return this.parent
+              .then(clear123DoneState());
           }
         })
         .then(function () {
           if (options['321done']) {
-            return clear123DoneState(this.parent, true);
+            return this.parent
+              .then(clear123DoneState( { untrusted: true }));
           }
         });
     };
   }
 
-  function clearContentServerState(context, options) {
+  function clearContentServerState(options) {
     options = options || {};
-    // clear localStorage to avoid polluting other tests.
-    var remote = getRemote(context);
-    return remote
-      // always go to the content server so the browser state is cleared,
-      // switch to the top level frame, if we aren't already. This fixes the
-      // iframe flow.
-      .switchToFrame(null)
-      .setFindTimeout(config.pageLoadTimeout)
-      .getCurrentUrl()
-      .then(function (url) {
-        // only load up the content server if we aren't
-        // already at the content server.
-        if (url.indexOf(CONTENT_SERVER) === -1 || options.force) {
-          return remote.get(require.toUrl(CONTENT_SERVER + 'clear'))
-                    .setFindTimeout(config.pageLoadTimeout)
-                    .findById('fxa-clear-storage-header');
-        }
-      })
+    return function () {
```

Comment:
Could implement these by wrapping in `thenify` again, as in another of your PRs from last week (forget which). No biggie, but if it does the job...